site/content/integrate/apps/interactivity/dynamic-form-example/_index.md

```diff
@@ -0,0 +1,544 @@
+---
+title: "Dynamic Form Example"
+heading: "Dynamic Form Example"
+description: "This guide explains how to provide dynamic values to forms, as well as how to show a new or modified form upon submission"
+weight: 5
+---
+
+### Overview
+
+Each of these examples will be using a modal form to help the user create a ticket in an external system such as Jira or Zendesk. The form will get a little more complex in each example. In this guide we'll cover:
+
+- How to render an item in the post dot menu
+- How to open a modal when the post menu item is clicked
+- How to create a form with:
+    - Text fields
+    - Static select fields
+    - Dynamic autocomplete select fields
+- How to redefine the form based on user interaction
+- Using `context` and `expand`:
+    - How to access the associated post after interacting with the post menu
+    - How to access other entities in the user's context such as the current `Team` or `Channel`
+
+### Setup
+
+These examples assume you have a Node.js App set up using [express](https://github.com/expressjs/express), like the [quick start example]({{< ref "quick-start-js" >}}), or the [example](https://github.com/mattermost/mattermost-plugin-apps/blob/master/dev/node_app/src/app.ts) from the App framework [development environment](https://github.com/mattermost/mattermost-plugin-apps/tree/master/dev).
+
+The examples on this page use types from the [mattermost-redux](https://github.com/mattermost/mattermost-redux) library, namely the App-specific types in [types/apps.ts](https://github.com/mattermost/mattermost-redux/blob/master/src/types/apps.ts); Mattermost types `Post`, `UserProfile`, and `Team`; and the Mattermost JavaScript client. See the end of the guide for the import paths for each of these.
```

Comment:
You talk about the types imported, but do not talk about importing also the Client4 library.